web/app/scripts/src/services/calculate-percentage.es.js

```diff
@@ -15,12 +15,17 @@ const module = angular.module(moduleProperties.moduleName, []);
 
 module.service(moduleProperties.componentName, [
     function () {
+        let qKeys = [];
```

Comment:
I would replace `countQuestionsTotal` with a function that finds the important questions (e.g. what you have but define `qKeys` within the function and return `qKeys` from the function:

```
        this.ImportQuestionKeys = function (form) {
            let qKeys = [];
            angular.forEach(form, function (question) {
                angular.forEach(question.fieldGroup, function (field) {
                    if (field.templateOptions.fieldOptions.important === true) {
                        qKeys.push(field.key)
                    }
                });
            });

            return qKeys;
        };
```

You can then simply get the count with `countQuestions = ImportQuestionKeys(form).length`

And you can pass the `qKeys` returned by `ImportQuestionKeys` to  `countNumberOfAnsweredQuestions` to avoid storing the `qKeys` in the service.